### PR TITLE
Minimize Library Entry Points with Client Component Focus

### PR DESCRIPTION
**Pull request by hobbescodes**
##### Task link: https://trello.com/c/zmRNXJwc/224-refactor-library-structure-with-client-component-focus

Update library structure to minimize entry points by putting focus on client components. New entry points are as follows:

1) `index.ts` (main entry point for root cascade layers, backfills, preset, components, and hooks).
2) `index.next.ts` (entry point for NextJS optimized components)

## Test Steps

1) Verify ALL component and recipe structures seem sound (emphasis on new recipe pattern to utilize `@ark-ui/anatomy`).
2) Verify that components render appropriately in storybook and downstream example apps.
3) verify that all tests pass.

### CODE REVIEW

File: .changeset/tricky-spoons-rule.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@animareflection/ui": minor
+---
+
+Refactor library structure with client component focus
```
Comment (coopbri) [inline .changeset/tricky-spoons-rule.md:5]: This is something we can discuss: what are your thoughts on known-breaking changes, such as this one, being prefixed with "**Breaking:**" ? Example: https://github.com/wagmi-dev/wagmi/blob/main/packages/core/CHANGELOG.md#minor-changes-4

Note also we can do this at the release PR level, just bringing this up so we have the decision formed moving forward. Personally, I think it's not strictly necessary, but might be nice. Let me know what you think.

Comment (hobbescodes) [reply .changeset/tricky-spoons-rule.md:5]: Yeah that sgtm! I think it would be nice as well.

**Comment by Twonarly1** [comment]
When using the `"@ark-ui/anatomy"` for the recipes, will this still allow us to create one-off slots for some use cases, or does it constrain our slots to what's preset?

**Comment by hobbescodes** [comment]
> When using the `"@ark-ui/anatomy"` for the recipes, will this still allow us to create one-off slots for some use cases, or does it constrain our slots to what's preset?

We can still create one-offs! Checkout the `NumberInput` recipe for a good example!

**Review by coopbri — APPROVED** [review]
Thanks for this big refactor, very clean and we can incrementally adopt RSCs/library server support down the line. Great work!!

**Review by Twonarly1 — APPROVED** [review]
Looks great! Storybook tests are all passing and confirmed the downstream apps are also rendering appropriately.

File: .changeset/tricky-spoons-rule.md (hunk 1 above)
Comment (coopbri) [reply .changeset/tricky-spoons-rule.md]: @Twonarly1 you probably saw this discussion but tagging just in case

**Pull request merged**